### PR TITLE
Feature/preview function

### PR DESCRIPTION
**Pull request by MHove89**
Sitecore 8 connector and preview functionality.

### CODE REVIEW

**Comment by emilras** [comment]
@MHove89 Awesome! We will review it and get it merged :)

File: src/Enterspeed.Source.SitecoreCms.V8/Services/EnterspeedConfigurationService.cs [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+        public List<EnterspeedSitecoreConfiguration> GetConfiguration()
+        {
+            Item enterspeedConfigurationItem = _itemManager.GetItem(EnterspeedIDs.Items.EnterspeedConfigurationID, Language.Parse("en"), Version.Latest, _factory.GetDatabase("web"));
+            if (enterspeedConfigurationItem == null || enterspeedConfigurationItem.Versions.Count == 0)
+            {
+                return new List<EnterspeedSitecoreConfiguration>();
+            }
+
+            //string enabled = enterspeedConfigurationItem[EnterspeedIDs.Fields.EnterspeedEnabledFieldID] ?? string.Empty;
```
Comment (elglogins) [inline src/Enterspeed.Source.SitecoreCms.V8/Services/EnterspeedConfigurationService.cs]: If this commented out part is irrelevant, can you please remove it?

File: src/Enterspeed.Source.SitecoreCms.V8/Services/EnterspeedConfigurationService.cs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                            if (languageEnterspeedSiteConfigurationItem.Versions.Count == 0)
+                            {
+                                languageEnterspeedSiteConfigurationItem = enterspeedSiteConfigurationItem;
+                            }
+
+                            Item homeItem = _itemManager.GetItem(siteContext.StartPath, siteLanguage, Version.Latest, siteContext.Database);
+
+                            if (homeItem == null || homeItem.Versions.Count == 0)
+                            {
+                                // TODO - KEK: throw exception here?
```
Comment (elglogins) [inline src/Enterspeed.Source.SitecoreCms.V8/Services/EnterspeedConfigurationService.cs]: Do you think we should throw an exception here?

**Review by elglogins — DISMISSED** [review]

File: src/Enterspeed.Source.SitecoreCms.V8/Services/EnterspeedConfigurationService.cs (hunk 1 above)
Comment (MHove89) [reply src/Enterspeed.Source.SitecoreCms.V8/Services/EnterspeedConfigurationService.cs]: @elglogins I have pushed the changes to the branch.

File: src/Enterspeed.Source.SitecoreCms.V8/Services/EnterspeedConfigurationService.cs (hunk 2 above)
Comment (MHove89) [reply src/Enterspeed.Source.SitecoreCms.V8/Services/EnterspeedConfigurationService.cs]: @elglogins I have pushed the changes to the branch.

**Review by elglogins — DISMISSED** [review]

**Pull request merged**